### PR TITLE
feat: Add scripture comment component

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

image

Adds an AddComment component. 

Things this PR handles
- Switches to an input when tapped. 
- Fires a user provided callback when the input is submitted. 

Things this doesn't handle 
- Tracking the input with the keyboard
- A "comment submitted" state. Right now I default to display a user's comment back to them. Might need to change this to a "Thanks for the comment" text.

### How do I test this PR?

Visit the AddCommentInput component via the Storybook.

### What automated tests did you write?

Snapshots over UI. 

## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-ui-kit/src/AddCommentInput/AddCommentInput.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import { H4, H5, BodySmall } from '../typography';
+import Touchable from '../Touchable';
+import Avatar from '../Avatar';
+import { withTheme } from '../theme';
+import PaddedView from '../PaddedView';
+
+const CommentAvatar = withTheme(
+  ({ theme: { sizing, colors } }) => ({
+    themeSize: sizing.baseUnit * 3,
+    buttonIcon: 'chunky-plus',
```
Comment (jimmyhogoboom) [inline packages/apollos-ui-kit/src/AddCommentInput/AddCommentInput.js:14]: Mild nitpick: Do we want an avatar with a plus here? My initial instinct when seeing that is to click the avatar to add photo or something like that, specific to the avatar.

**Review by jimmyhogoboom — APPROVED** [review]
Looks good to me! Just had the one small question/nitpick, otherwise seems good to go.

File: packages/apollos-ui-kit/src/AddCommentInput/AddCommentInput.js (hunk 1 above)
Comment (vinnyjth) [reply packages/apollos-ui-kit/src/AddCommentInput/AddCommentInput.js:14]: That's a great question! @conrad-vanl, do you have any thoughts? I implemented what you had in Figma (screenshots in PR description)
